onmt/modules/Encoder.lua

```diff
@@ -206,6 +226,10 @@ function Encoder:forward(batch)
   end
   if self.train then
     self.inputs = {}
+    self.network:apply(function(m) if m.noiseInit then m.noiseInit[1]=0 end end)
+    if self.args.dropout_words and self.args.dropout_words > 0 then
+      onmt.VDropout.dropoutWords(self.args.dropout_words, batch)
```

Comment:
it is actually one part of the variational dropout but you are right, it can/should be part of the batch preparation.